src/tools/compiletest/src/runtest.rs

```diff
@@ -309,10 +316,18 @@ impl<'test> TestCx<'test> {
         }
     }
 
+    fn effective_pass_mode(&self) -> Option<PassMode> {
+        if !self.props.ignore_pass {
+            if let (mode @ Some(_), Some(_)) = (self.config.force_pass_mode, self.props.pass_mode) {
+                return mode;
+            }
+        }
+        self.props.pass_mode
+    }
+
     fn should_run_successfully(&self) -> bool {
         match self.config.mode {
-            RunPass => true,
-            Ui => self.props.pass_mode == Some(PassMode::Run),
+            RunPass | Ui => self.effective_pass_mode() == Some(PassMode::Run),
```

Comment:
The old code is correct, `RunPass` tests are always "run".
They may have annotations due to tests being moved around in the past, but those annotations are ignored.
(This doesn't matter much though given that the run-pass tests suite is going to be removed soon.)

Reply:
Hmm; I thought we wanted `--run check` to affect the `src/test/run-pass` as well?

Reply:
Ah, ok, the mode from `--pass` should indeed be respected.
Then in-source annotations can be respected as well, this is all temporary anyway.